pkg/reconciler/apis/apiexport/apiexport_reconcile.go

```diff
@@ -110,14 +109,16 @@ func (c *controller) reconcile(ctx context.Context, apiExport *apisv1alpha1.APIE
 }
 
 func (c *controller) ensureSecretNamespaceExists(ctx context.Context, clusterName logicalcluster.Name) {
-	logger := klog.FromContext(ctx).WithValues("Namespace", clusters.ToClusterAwareKey(clusterName, c.secretNamespace))
+	logger := klog.FromContext(ctx)
 	ctx = klog.NewContext(ctx, logger)
 	if _, err := c.getNamespace(clusterName, c.secretNamespace); errors.IsNotFound(err) {
 		ns := &corev1.Namespace{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: c.secretNamespace,
+				Name:        c.secretNamespace,
+				Annotations: map[string]string{logicalcluster.AnnotationKey: clusterName.String()},
```

Comment:
Just making sure: this is only needed for `logging.WithObject` to include the cluster name?

Reply:
Yes. I'm open to better ideas :|